### ISSUE TITLE
Uppercase "V" should be supported for TargetFrameworkVersion

### ISSUE THREAD

**Issue by dsplaisted** [NuGet/Home#6823]
- Create a new .NET Framework project (not SDK-style)
- Modify `TargetFrameworkVersion` to have an uppercase `V`, e.g. `<TargetFrameworkVersion>V4.6.1</TargetFrameworkVersion>`
- Run `msbuild /restore`

**Expected:** Restore succeeds
**Actual:**

```
C:\git\dotnet-sdk\artifacts\.dotnet\2.1.300-preview3-008414\sdk\2.1.300-preview3-008414\NuGet.targets(114,5): Invalid framework version 'V4.6.1'. [C:\git\dotnet-sdk\artifacts\Debug\tmp\It_allows_Tar---0EA22DDA\UppercaseTargetFrameworkVersion\UppercaseTargetFrameworkVersion.csproj]
```

**Note:** This requires tools with fixes for the following issues, otherwise other errors may be generated before NuGet has a chance to error here:

- https://github.com/dotnet/sdk/issues/1884
- https://github.com/Microsoft/msbuild/issues/2886

**Comment by nkolev92** [comment]
We want this for 2.1.3xx?
Is this really escrow worthy?

//cc
@rrelyea

**Comment by rohit21agrawal** [comment]
2.1.3xx is a different shipping cycle, you don't have to go through escrow for that.

**Comment by nkolev92** [comment]
I think we still do because we're not inserting into 2xx and 3xx independently.
We'll not insert 4.8 bits in 2.1.3xx because it'll ship way earlier than 4.8 stabilizes. 
We'd need to insert into 4.7 then, which means it effectively has to go in VS 15.7 and 2.1.2xx.

**Comment by rohit21agrawal** [comment]
No, I already talked to matt, we can insert into 2.1.3xx as and when we like. I already have pretty recent dev bits flowing into 2.1.3xx

**Comment by mishra14** [comment]
@dsplaisted How can I get the latest sdk/msbuild changes to test the nuget fix?

**Comment by dsplaisted** [comment]
@mishra14 It should be in VS 15 Update 7

**Comment by rrelyea** [comment]
Unsure if we target 15.8, 15.9 or 16.0 with this fix. Keeping in 4.8 milestone for now, but we'll decide where to back port it after we have fix.

**Comment by nkolev92** [comment]
We have not done work for this in a while. 

@dsplaisted If this is a pressing issue, please let us know so we can re-prioritize.
